### PR TITLE
[HDInsight] Add two commands to support hdinsight node reboot feature

### PR DESCRIPTION
**Pull request by aim-for-better**
**Description**  
Add two commands: 
- `az hdinsight host list` to support list the hosts of the HDInsight cluster
- `az hdinsight host restart` to support restart specific hosts of the HDInsight cluster

**Testing Guide**  

**History Notes**  

`az hdinsight host list` to support list the hosts of the HDInsight cluster
`az hdinsight host restart` to support restart specific hosts of the HDInsight cluster

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
HDInsight

**Comment by aim-for-better** [comment]
> Could you add tests for these two commands?

Hi @MyronFanQiu  I have added the related tests. Could you please review again?

File: src/azure-cli/azure/cli/command_modules/hdinsight/_params.py [hunk 1]
```diff
@@ -237,3 +237,9 @@ def load_arguments(self, _):
         c.argument('primary_key', help='The certificate for the Log Analytics workspace. '
                                        'Required when workspace ID is provided.')
         c.ignore('workspace_type')
+
+    with self.argument_context('hdinsight host') as c:
+        c.argument('cluster_name', options_list=['--cluster-name'],
+                   completer=get_resource_name_completion_list('Microsoft.HDInsight/clusters'),
+                   help='The name of the cluster.')
+        c.argument('hosts', nargs="+", help='The host name of the cluster.')
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/hdinsight/_params.py:245]: The name of this argument is a little bit confusing for me. Should we add `options_list` for it?

**Comment by MyronFanQiu** [comment]
@haroldrandom @Juliehzl Please review this PR as the code onwer.

File: src/azure-cli/azure/cli/command_modules/hdinsight/_params.py (hunk 1 above)
Comment (aim-for-better) [reply src/azure-cli/azure/cli/command_modules/hdinsight/_params.py]: > The name of this argument is a little bit confusing for me. Should we add `options_list` for it?

Updated

File: src/azure-cli/azure/cli/command_modules/hdinsight/commands.py [hunk 2]
```diff
@@ -92,3 +98,9 @@ def load_command_table(self, _):
         g.show_command('show', 'get_monitoring_status')
         g.custom_command('enable', 'enable_hdi_monitoring')
         g.command('disable', 'disable_monitoring')
+
+    # VirtualMachine operations
+    with self.command_group('hdinsight host', hdinsight_virtual_machines_sdk,
+                            client_factory=cf_hdinsight_virtual_machines) as g:
+        g.command('list', 'list_hosts')
+        g.command('restart', 'restart_hosts', confirmation=True, supports_no_wait=True)
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/hdinsight/commands.py:106]: If support no wait, could we add wait command to wait till condition matched?

Comment (aim-for-better) [reply src/azure-cli/azure/cli/command_modules/hdinsight/commands.py:106]: > If support no wait, could we add wait command to wait till condition matched?

Hi @Juliehzl  Thanks for reminder. I have added the wait command.

Comment (aim-for-better) [reply src/azure-cli/azure/cli/command_modules/hdinsight/commands.py:106]: > > If support no wait, could we add wait command to wait till condition matched?
> 
> Hi @Juliehzl Thanks for reminder. I have added the wait command.

Hi @Juliehzl  As we talked I removed `support_no_wait`, Thank you very much!

**Pull request merged**